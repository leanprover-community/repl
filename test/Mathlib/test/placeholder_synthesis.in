{"cmd": "import Mathlib"}

{"cmd": "example : 1 = 0 := by\n  cases' exists_ne 2 with n h\n  simpa [] using h _", "env": 0}

{"cmd": "example : 1 = 0 := sorry", "env": 0, "sorries": "individual"}

{"tactic": "cases' exists_ne 2 with n h", "proofState": 0}

{"tactic": "simpa [] using h _", "proofState": 1}
