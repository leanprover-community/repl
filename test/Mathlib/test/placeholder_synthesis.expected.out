{"env": 0}

{"messages":
 [{"severity": "error",
   "pos": {"line": 3, "column": 19},
   "endPos": {"line": 3, "column": 20},
   "data":
   "don't know how to synthesize placeholder\ncontext:\nn : ℕ\nh : n ≠ 2\n⊢ n = 2"}],
 "env": 1}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 19},
   "parentDecl": "_example",
   "goals": ["⊢ 1 = 0"],
   "endPos": {"line": 1, "column": 24}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 2}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals": ["case intro\nn : ℕ\nh : n ≠ 2\n⊢ 1 = 0"]}

{"proofStatus": "Incomplete: contains metavariable(s)",
 "proofState": 2,
 "goals": []}

